xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-api/src/main/java/org/xwiki/security/authentication/RegistrationConfiguration.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.security.authentication;
+
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Define the various configuration options for registration.
+ *
+ * @version $Id$
+ * @since 15.10RC1
+ */
+@Unstable
+@Role
+public interface RegistrationConfiguration
+{
+    /**
+     * Default minimum password length.
+     */
+    int DEFAULT_MINIMUM_PASSWORD_LENGTH = 6;
+
+    /**
+     * Represents the rules to apply for creating a new password.
+     */
+    enum PasswordRules
+    {
+        /**
+         * When one lower case character is mandatory.
+         */
+        ONE_LOWER_CASE_CHARACTER(".*[a-z]+.*"),
+
+        /**
+         * When one upper case character is mandatory.
+         */
+        ONE_UPPER_CASE_CHARACTER(".*[A-Z]+.*"),
+
+        /**
+         * When one symbol character is mandatory.
+         */
+        ONE_SYMBOL_CHARACTER(".*(_\\W)+.*"),
+
+        /**
+         * When one number character is mandatory.
+         */
+        ONE_NUMBER_CHARACTER(".*[0-9]+.*");
```

Comment:
That's probably another discussion but basing the password strength on the presence of some type of chars is not really state of the art. Defining a minimal password length would be a more secure alternative for instance.

Reply:
You can do both.